Azure-Samples/azure-ml-federated-learning
# FL native contract backed by the factory engine

## Purpose

* Replaced the factory code's `submit.py` with the actual FL native contract.
* Aligned the `config.yaml` file to the spec CLI YAML file.
* The factory engine backs the pipeline.

## What is the expected review turnaround time?

Urgency:
- [ ] High (needs review today)
- [x] Medium (needs review within a few days - most common)
- [ ] Low (can wait a week)

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

Submit the factory pipeline and see if everything is as expected.

MNIST run: https://ml.azure.com/experiments/id/cd48cb1d-2cee-4b10-8a76-2d094b2a5868/runs/patient_kite_n53jmp1kv6?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/amitgarg-fldev-rg/workspaces/aml-testfactory&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
